### PR TITLE
feat: Add cohort sync

### PR DESCRIPTION
**Pull request by zhukaihan**
### Summary

Add cohort targeting and download to SDK. 
GH Action changes: 
* Using LTS node versions: 16, 18, 20, 22, 24. All are in either maintenance or active LTS. 
* Added env secrets. 

### Tests
Unit tests created for all added functions. 
Integration test created for user id and group id and group name cohort targeting for US endpoint. 
Integration test for EU endpoint contains simple cohort targeting. 
Manually tested cohort syncs at flag update interval when cohort was updated. 

### Checklist

* [x] Does your PR title have the correct [title format](https://github.com/amplitude/experiments-js-server/blob/main/CONTRIBUTING.md#pr-commit-title-conventions)?
* Does your PR have a breaking change?:  no

### CODE REVIEW

File: packages/node/src/local/client.ts [hunk 1]
```diff
@@ -5,6 +5,9 @@ import {
   topologicalSort,
 } from '@amplitude/experiment-core';
 import EventSource from 'eventsource';
+import { USER_GROUP_TYPE } from 'src/types/cohort';
```
Comment (tyiuhc) [inline packages/node/src/local/client.ts:8]: imports (in non-testcase files) should be relative

File: packages/node/src/local/updater.ts [hunk 2]
```diff
@@ -24,3 +30,141 @@ export interface FlagConfigUpdater {
...
+      if (!Object.is(current, flagConfigs)) {
+        changed = true;
+      }
+    }
+
+    // Get all cohort needs update.
+    const cohortIds = CohortUtils.extractCohortIds(flagConfigs);
+    if (cohortIds && cohortIds.size > 0 && !this.cohortFetcher) {
+      throw Error(
+        'cohort found in flag configs but no cohort download configured',
```
Comment (tyiuhc) [inline packages/node/src/local/updater.ts:71]: Good that this is checked - but what if they want to use only a subset of flags in a deployment, and that subset doesn't require cohorts 🤔 (cc: @bgiori)

File: packages/node/src/local/cohort/poller.ts [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+  public async start(
+    onChange?: (storage: CohortStorage) => Promise<void>,
+  ): Promise<void> {
+    if (!this.poller) {
+      this.logger.debug('[Experiment] cohort poller - start');
+      this.poller = setInterval(async () => {
+        try {
+          await this.update(onChange);
+        } catch (e) {
+          this.logger.debug('[Experiment] flag config update failed', e);
```
[the remark this answers is not in the document]
Comment (zhukaihan) [reply packages/node/src/local/cohort/poller.ts:46]: That error is by jest, which test should complete within 5000 ms. I think the root cause is cohort download have not completed. I need to add `cohortRequestDelayMillis` config to define delay between retries. As well as tests for incorrect api keys.

**Review by bgiori — APPROVED** [review]
Hold off on merging yet

File: packages/node/src/types/config.ts [hunk 4]
```diff
@@ -210,3 +246,18 @@ export const LocalEvaluationDefaults: LocalEvaluationConfig = {
 export const AssignmentConfigDefaults: Omit<AssignmentConfig, 'apiKey'> = {
   cacheCapacity: 65536,
 };
+
+export const CohortConfigDefaults: Omit<CohortConfig, 'apiKey' | 'secretKey'> =
+  {
+    cohortServerUrl: 'https://cohort-v2.lab.amplitude.com',
+    maxCohortSize: 10_000_000,
```
Comment (bgiori) [inline packages/node/src/types/config.ts:253]: max cohort size should be 2147483647

**Review by tyiuhc — APPROVED** [review]

**Pull request merged**